Automattic/wp-calypso
# Checkout: Use dropdown for wpcom variant picker and remove first step edit

#### Proposed Changes

Some products have term variants (monthly, yearly, biennial) which are displayed in the first step during checkout. Because these take up a large amount of vertical height (particularly when there are multiple products in the cart), they were originally placed behind an "Edit" button for that step along with other cart-changing behavior like product removal buttons. (Worth noting that the first step isn't part of the actual array of checkout steps; it is a manually managed step that requires a lot of machinery to operate.)

Over time, the cart-changing behavior has been moved so that it's accessible outside of the "Edit" mode on the first step. Right now the only remaining difference between the step's active and inactive modes is the variant picker.

Not that long ago, the Jetpack checkout was modified to replace their variant picker with a drop-down to take up less vertical space; this allowed it to be displayed in the step's inactive mode and allowed removing the "Edit" mode entirely.

In this diff, we modify the wpcom version of checkout to do the same thing. Removing the "Edit" mode will save a considerable amount of code and should decrease the complexity for users.

The major downside of this change is that the user cannot see all the variations at once, and only sees the savings percentage when a variant is selected, but the major upside is that the user can see the variant picker without having to click "Edit".

#### Screenshots

Before:

<img width="567" alt="Screen Shot 2022-07-07 at 12 31 10 PM" src="https://user-images.githubusercontent.com/2036909/177824771-8c4ce242-84f3-44b7-b788-d57847f72f05.png">

<img width="562" alt="Screen Shot 2022-07-07 at 12 31 17 PM" src="https://user-images.githubusercontent.com/2036909/177824795-197ccf50-a8af-438d-94d9-7090d63f5dc1.png">

After:

<img width="566" alt="Screen Shot 2022-07-07 at 12 32 20 PM" src="https://user-images.githubusercontent.com/2036909/177824822-3e2ee539-43aa-457b-8b20-334c4daacc78.png">


#### Testing Instructions

- Add a wpcom product to your cart that has term variants. For now you may need to use legacy plans because there are no variants for the Starter or Pro plans yet. You can add a legacy product to your cart by URL. For example, visit `/checkout/premium` to add a Premium plan to your cart.
- Verify that you see the term variant picker for the plan without having to change any modes.
- Verify that there is no "Edit" button on the first step.
